### PR TITLE
Add password length validation - take 2

### PR DESCRIPTION
**Pull request by hwwhww**
Address #99
Replace #118

I haven't figured out how to test hidden input (password) with the test script, but #123 gave me a hint of utilizing click callback function. This is a workaround of #118.

### Example output
1. enter the length=3 password and confirmation -> got rejected
2. enter the length=5 password -> got rejected
3. enter the length=8 password and confirmation -> got accepted

```sh
. venv/bin/activate && python ./eth2deposit/deposit.py
Please choose how many validators you wish to run: 1
Please choose your mnemonic language (czech, chinese_traditional, chinese_simplified, english, spanish, italian, korean) [english]:
Please choose the (mainnet or testnet) network/chain name (mainnet, witti, altona, medalla, spadina, zinken) [mainnet]:
Type the password that secures your validator keystore(s):
Repeat for confirmation:
Error: The password length should be at least 8. Got 3. Please retype again.
Type the password that secures your validator keystore(s):
Error: The password length should be at least 8. Got 5. Please retype again.
Type the password that secures your validator keystore(s):
Repeat for confirmation:
```

### Compare with #118

|    | #118                                                            | This PR                                                                                                                                                      |
|----|-----------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------|
| 1. | `--password` flag is disallowed                                    | `--password` flag is allowed                                                                                                                              |
| 2. | Validate the password strength once the user enters the password | Validate the password strength  *after* it passing the confirmation. After that, it validates the strength once the user enters the password. (another while loop) |

I do want to avoid (1), but (1) helps us writing the test script. 😓

### CODE REVIEW

**Review by CarlBeek — APPROVED** [review]
I think this is a better solution overall than the one presented in #118. 👍 

I do with the order was verify length then check for matching passwords, but this may be an example of perfect being the enemy of the good.

**Comment by hwwhww** [comment]
edited PR description:

#118: `--password` flag is disallowed
This PR: `--password` flag is allowed

File: eth2deposit/deposit.py [hunk 1]
```diff
@@ -51,6 +55,38 @@ def check_python_version() -> None:
...
+
+
+def validate_password(cts: click.Context, param: Any, password: str) -> str:
+    is_valid_password = False
+
+    # The given password has passed confirmation
+    try:
+        validate_password_strength(password)
+    except ValidationError as e:
+        click.echo(f'Error: {e} Please retype again.')
```
Comment (hwwhww) [inline eth2deposit/deposit.py:69]: please retype again.
```suggestion
        click.echo(f'Error: {e} Please retype.')
```

File: eth2deposit/deposit.py [hunk 2]
```diff
@@ -51,6 +55,38 @@ def check_python_version() -> None:
...
+    else:
+        is_valid_password = True
+
+    while not is_valid_password:
+        password = get_password(text='Type the password that secures your validator keystore(s)')
+        try:
+            validate_password_strength(password)
+        except ValidationError as e:
+            click.echo(f'Error: {e} Please retype again.')
```
Comment (hwwhww) [inline eth2deposit/deposit.py:78]:
```suggestion
            click.echo(f'Error: {e} Please retype.')
```

**Pull request merged**